mongo/mongo_test.go

```diff
@@ -120,16 +120,25 @@ func TestMongoHelpers(t *testing.T) {
 		})
 	})
 	t.Run("transform aggregate pipeline", func(t *testing.T) {
```

Comment:
We should probably have some tests for more complicated pipelines. For example, it's not super useful to use a pipeline that just contains $out and verify that `hasOutputStage` is true because that doesn't validate that we're actually checking the _last_ stage in the pipeline as well.

Reply:
Sounds good. Added two more test cases with `$out` in various positions.